lib/ret_web/controllers/api/v1/avatar_controller.ex

```diff
@@ -22,7 +25,23 @@ defmodule RetWeb.Api.V1.AvatarController do
   end
 
   def create(conn, %{"avatar" => params}) do
-    create_or_update(conn, params, %Avatar{})
+    avatar =
+      with %{"parent_avatar_listing_id" => sid} when not is_nil(sid) and sid != "" <- params,
+           parent_listing <-
```

Comment:
probably worth extracting this somewhere else, particularly if it can get out of the controller.